### PR TITLE
Plasma3 d

### PR DESCRIPTION
**Pull request by wingena**
Latest updates to plasma3D

### CODE REVIEW

File: docker/Dockerfile
Comment (tloobyCFS) [inline docker/Dockerfile:44]: test comment for @wingena

**Comment by tloobyCFS** [comment]
nice.  merging main into plasma3D was a good idea to fix the issues we saw

File: source/plasma3DClass.py
Comment (plasmapotential) [inline source/plasma3DClass.py:57]: @wingena in allowed_class_vars() there should be a markdown compatible comment that describes each variable.  This comment is not only useful for future users, but also is used directly in the readthedocs to describe the parameters needed in the input file.  If there are units associated with an input variable, they should be defined here.  Here is an example code snippet to get you started:

```
   def allowed_class_vars(self):
      .. Writes a list of recognized class variables to HEAT object
      .. Used for error checking input files and for initialization

       plasma3D Variables:
       ----------------------------

      :plasma3Dmask:  boolean that determines if 3D plasmas should be turned on.  If True, uses M3DC1 equilibrium
      :itt: integer number of toroidal iterations...etc.

etc. etc...
```

I added your plasma3DClass allowedClassVars to the documentation in this commit:  https://github.com/plasmapotential/HEAT/pull/31/commits/d9eabb7bdd8371274927b2a2b4df4ee19e3549e0

This is where the input file parameters from allowedClassVars are rendered:  https://heat-flux-engineering-analysis-toolkit-heat.readthedocs.io/en/latest/inputFile.html



Please go through the allowedClassVars and add a separate line for each variable.  For a more extensive example, see the allowedClassVars function in heatFluxClass.py

File: source/plasma3DClass.py
Comment (plasmapotential) [inline source/plasma3DClass.py:537]: @wingena same thing here.  please add a markdown compatible description for each variable here.  Additionally, you will need to add the following line to  docs/source/inputFile.rst:
```
.. autofunction:: plasma3DClass.heatflux3D.allowed_class_vars  
```
in order to get your comments added to the readthedocs.  I only did the plasma3D class in my commit, so you will need to add the line above to the inputFile.rst on your own (a good exercise to understand how it works).

File: tests/integrationTests/D3DTestCase/batchFile.dat
Comment (plasmapotential) [inline tests/integrationTests/D3DTestCase/batchFile.dat:8]: @wingena this comment reflects the old (v3.0) HEAT batchFile.  Copy the new comment from the NSTXU test case [here](https://github.com/plasmapotential/HEAT/blob/f6b7d74b10e56f0bb458f2bc4cec990e80a039df/tests/integrationTests/nstxuTestCase/batchFile.dat).

More specifically, the shot and timestep columns are missing.

**Comment by plasmapotential** [comment]
@wingena this looks pretty good!  I added a few minor comments throughout the code that should be addressed before we merge the branch into main.  Generally they are just CI and workflow related comments.  

Also, note that I added plasma3D to be the target for CI in this commit: https://github.com/plasmapotential/HEAT/pull/31/commits/3373d465f706532662f95d5d7545810e631be893 .  This means that when you push your code up to github it will automatically trigger the integration tests (only for NSTXU currently until I rebuild docker container with M3DC1 files for D3D as we discussed).  You can see the status of the tests here on the PR page, or [here](https://github.com/plasmapotential/HEAT/actions).  

Be sure to "pull" the plasma3D branch down to your machine to get my updates before you go through and edit things.  Once you are done resolving each comment, mark it as resolved.  When you are ready for me to review again, write a comment here or ping me via email and I will do the final merge into main.

File: source/plasma3DClass.py
Comment (wingena) [reply source/plasma3DClass.py:57]: Thanks, I updated the variable description

File: source/plasma3DClass.py
Comment (wingena) [reply source/plasma3DClass.py:567]: Same here, I updated it.

File: tests/integrationTests/D3DTestCase/batchFile.dat
Comment (wingena) [reply tests/integrationTests/D3DTestCase/batchFile.dat:7]: Thanks, I updated the comments in batch file

**Comment by wingena** [comment]
@plasmapotential I resolved the comments you made. The push triggered the test and passed.

**Review by plasmapotential — APPROVED** [review]
approved plasma3D after comments were resolved

**Pull request merged**
